### ISSUE TITLE
Can we use C++17?

### ISSUE THREAD

**Issue by mattinmir** [HPCE/hpce-2016-cw6#8]
The makefile has option `std=c++11`: does this mean we are limited to C++11 or can we use features from later versions?

**Comment by m8pple** [comment]
c++11 is mostly a function of habit and platform compatibility.
I'm not up to date on c++17 support across platforms, nor
the completeness of that support.

That said, as long as it compiles in my default AWS instance g++, then
I'm happy with it.  I not, then I guess I could upgrade the instance
install (it is a bit old now anyway). I don't have a HPCE AMI instance open
right now - do you know if the g++ on there supports it?

In summary: yes, as long as it isn't too much hassle getting g++17 

What features of c++17 are you using, out of interest?

**Comment by mattinmir** [comment]
I'm not using any particular features at the moment, but want to know in advance in case there is a feature I decide to use. So the question includes asking about C++14 also.

I'll just take the answer to be 'whatever runs on the version of g++ on the AMI'

Thanks

**Comment by mattinmir** [comment]
For reference, the AMI has GCC 4.9.2, which at the very least supports C++1y, which is almost the same as C++14
